File: VOLUME 2/AGENDAR MENSAGENS/AGENDAR COM AVATAR.py
from telegram import Update, ParseMode
from telegram.ext import Updater, CommandHandler, MessageHandler, Filters, CallbackContext
import datetime

# Função para o comando /schedule
def schedule_message(update: Update, context: CallbackContext) -> None:
    # Obtém a mensagem a ser agendada a partir do argumento do comando
    message = ' '.join(context.args)

    # Pergunta ao usuário quando a mensagem deve ser enviada
    update.message.reply_text('Quando você quer que eu envie esta mensagem? Por favor, me diga a data e hora usando o formato "dd/mm/yyyy hh:mm".')

    # Armazena a mensagem e aguarda a resposta do usuário com a data e hora
    context.user_data['scheduled_message'] = message
    context.user_data['schedule_step'] = 'date_time'

# Função para a resposta do usuário com a data e hora para agendamento
def set_date_time(update: Update, context: CallbackContext) -> None:
    # Obtém a data e hora informada pelo usuário
    date_time_str = update.message.text.strip()

    # Tenta converter a data e hora para um objeto datetime
    try:
        date_time = datetime.datetime.strptime(date_time_str, '%d/%m/%Y %H:%M')
    except ValueError:
        update.message.reply_text('Desculpe, não entendi a data e hora. Por favor, tente novamente usando o formato "dd/mm/yyyy hh:mm".')
        return

    # Verifica se a data e hora informada é no futuro
    if date_time < datetime.datetime.now():
        update.message.reply_text('Desculpe, não posso agendar uma mensagem para o passado. Por favor, informe uma data e hora futuras.')
        return

    # Armazena a data e hora informada pelo usuário e aguarda a confirmação do grupo
    context.user_data['scheduled_date_time'] = date_time
    context.user_data['schedule_step'] = 'confirm'
    update.message.reply_text(f'Ok, vou enviar a mensagem "{context.user_data["scheduled_message"]}" em {date_time.strftime("%d/%m/%Y %H:%M")}. Por favor, confirme o agendamento com o comando /confirm_schedule.')

# Função para confirmar o agendamento com o grupo
def confirm_schedule(update: Update, context: CallbackContext) -> None:
    # Obtém a mensagem, a data e hora agendadas e o chat_id do usuário
    message = context.user_data['scheduled_message']
    date_time = context.user_data['scheduled_date_time']
    chat_id = update.message.chat_id

    # Envia uma mensagem para confirmar o agendamento
    context.bot.send_message(chat_id=chat_id, text=f'A mensagem "{message}" será enviada em {date_time.strftime("%d/%m/%Y %H:%M")}. Por favor, confirme o agendamento com o comando /confirm_schedule.')

# Função para enviar a mensagem agendada
def send_scheduled_message(context: CallbackContext) -> None:
    # Obtém a mensagem, a data e hora agendadas e o chat_id do usuário
    message = context.job.context['scheduled_message']
    chat_id = context.job.context['chat_id']

    # Envia a mensagem agendada para o grupo
    context.bot.send


# Para criar um bot do Telegram em Python que permita agendar mensagens usando o seu perfil em grupos, podemos usar a biblioteca python-telegram-bot. 

File: VOLUME 2/AGENDAR MENSAGENS/AGENDAR COM AVATAR.js
const Telegraf = require('telegraf')
const Extra = require('telegraf/extra')
const Markup = require('telegraf/markup')
const schedule = require('node-schedule')

// Cria uma nova instância do bot
const bot = new Telegraf(process.env.BOT_TOKEN)

// Armazena as mensagens agendadas
const scheduledMessages = {}

// Responde ao comando /schedule
bot.command('schedule', (ctx) => {
  // Obtém a mensagem a ser agendada a partir do argumento do comando
  const message = ctx.message.text.slice(9).trim()

  // Pergunta ao usuário quando a mensagem deve ser enviada
  ctx.reply('Quando você quer que eu envie esta mensagem? Por favor, me diga a data e hora usando o formato "dd/mm/yyyy hh:mm".')

  // Armazena a mensagem e aguarda a resposta do usuário com a data e hora
  scheduledMessages[ctx.from.id] = { message }
  bot.on('text', setDateTime)
})

// Responde à resposta do usuário com a data e hora para agendamento
function setDateTime(ctx) {
  // Obtém a data e hora informada pelo usuário
  const dateTimeStr = ctx.message.text.trim()

  // Tenta converter a data e hora para um objeto Date
  const dateTime = new Date(dateTimeStr)
  if (isNaN(dateTime.getTime())) {
    ctx.reply('Desculpe, não entendi a data e hora. Por favor, tente novamente usando o formato "dd/mm/yyyy hh:mm".')
    return
  }

  // Verifica se a data e hora informada é no futuro
  if (dateTime.getTime() < Date.now()) {
    ctx.reply('Desculpe, não posso agendar uma mensagem para o passado. Por favor, informe uma data e hora futuras.')
    return
  }

  // Armazena a data e hora informada pelo usuário e aguarda a confirmação do grupo
  const scheduledMessage = scheduledMessages[ctx.from.id]
  scheduledMessage.dateTime = dateTime
  ctx.reply(`Ok, vou enviar a mensagem "${scheduledMessage.message}" em ${dateTime.toLocaleString()}. Por favor, confirme o agendamento com o comando /confirm_schedule.`)

  // Remove o handler de data e hora
  bot.off('text', setDateTime)

  // Aguarda a confirmação do grupo
  bot.command('confirm_schedule', confirmSchedule)
}

// Responde ao comando /confirm_schedule
function confirmSchedule(ctx) {
  // Obtém a mensagem agendada e o chat_id do usuário
  const scheduledMessage = scheduledMessages[ctx.from.id]
  const chatId = ctx.chat.id

  // Agenda a mensagem
  const job = schedule.scheduleJob(scheduledMessage.dateTime, () => {
    ctx.telegram.sendMessage(chatId, scheduledMessage.message)
  })

  // Remove a mensagem agendada
  delete scheduledMessages[ctx.from.id]

  // Envia uma mensagem para confirmar o agendamento
  ctx.reply(`A mensagem "${scheduledMessage.message}" será enviada em ${scheduledMessage.dateTime.toLocaleString()}. O agendamento foi confirmado.`)
}

// Para criar um bot do Telegram em JavaScript que permita agendar mensagens usando o seu perfil em grupos, podemos usar a biblioteca node-telegram-bot-api.